precompiles/types/errors_test.go

```diff
@@ -46,3 +47,63 @@ func Test_ErrInvalidMethod(t *testing.T) {
 		t.Errorf("Expected %v, got %v", expect, got)
 	}
 }
+
+func Test_ErrInvalidCoin(t *testing.T) {
+	e := ErrInvalidCoin{
```

Comment:
Consider checking `errors.Is(err, ErrInvalidCoin{})` or smth similar

Reply:
Also, can you elaborate on a need for using structs per each error? We can use this approach:

```go
var ErrInvalidCoin = errors.New("invalid coin")

// ...

return errors.Wrap(ErrInvalidCoin, "reason")


```

Reply:
The reason is purely aesthetical: so I set explicitly the way errors and their parameters are going to be formatted.

Reply:
I like using typed errors whenever possible, but I think these tests should be refactored to use [require.ErrorIs](https://pkg.go.dev/github.com/stretchr/testify/require#ErrorIs) or [require.ErrorAs](https://pkg.go.dev/github.com/stretchr/testify/require#ErrorAs)

Reply:
Done in 82f5204a40820325cafba5755c3ee634344aa5ca